### PR TITLE
chore: add github action to update licenses

### PR DESCRIPTION
**Pull request by fd0r**

### CODE REVIEW

**Comment by fd0r** [comment]
I advise to merge as is so that I can further debug the action (an action needs to be merged on main to be launchable)

File: .github/workflows/update_licenses.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        with:
+          token: ${{ secrets.BOT_TOKEN }}
+
+      - name: Install dependencies
+        if: ${{ matrix.os == "ubuntu-20.04" }}
+        run: |
+          ./script/make_utils/setup_os_deps.sh --linux-install-python
+          make setup_env
+
+      - name: Install dependencies
```
Comment (RomanBredehoft) [inline .github/workflows/update_licenses.yaml:47]: you have two "install dependencies" steps !

File: .github/workflows/update_licenses.yaml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        if: ${{ matrix.os != "ubuntu-20.04" }}
+        run: |
+          ./script/make_utils/setup_os_deps.sh
+          make setup_env
+
+      - name: Set version
```
Comment (RomanBredehoft) [inline .github/workflows/update_licenses.yaml:53]: wrong step name (you only build licenses here)

File: .github/workflows/update_licenses.yaml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        run: |
+          make licenses
+
+      # TODO: set dependency hash in git branch and commit
+      # Open a PR with the new version and updated apidocs
```
Comment (RomanBredehoft) [inline .github/workflows/update_licenses.yaml:58]: wrong commment

**Review by RomanBredehoft — DISMISSED** [review]
sorry, I wanted to approve

File: .github/workflows/update_licenses.yaml (hunk 1 above)
Comment (fd0r) [reply .github/workflows/update_licenses.yaml:47]: I know, one is for Mac and the other for linux

File: .github/workflows/update_licenses.yaml (hunk 2 above)
Comment (fd0r) [reply .github/workflows/update_licenses.yaml:53]: oups yes bad copy-paste

File: .github/workflows/update_licenses.yaml (hunk 3 above)
Comment (fd0r) [reply .github/workflows/update_licenses.yaml:58]: indeed

File: .github/workflows/update_licenses.yaml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        if: ${{ matrix.os != "ubuntu-20.04" }}
+        run: |
+          ./script/make_utils/setup_os_deps.sh
+          make setup_env
+
+      - name: Set version
+        run: |
+          make licenses
+
+      # TODO: set dependency hash in git branch and commit
```
[the remark this answers is not in the document]
Comment (fd0r) [reply .github/workflows/update_licenses.yaml:57]: I'll add a fixme

**Comment by fd0r** [comment]
I'll fix the changes so that at least the start is clean.

Then I'll check how it goes and fix if necesary

**Review by RomanBredehoft — APPROVED** [review]
thanks a lot ! happy debugging

**Comment by bcm-at-zama** [comment]
So, how does it work? Suppose you create a new branch new_branch, add a dependancy: the PR is red if you don't update the licenses there. Then, you launch your action directly in new_branch?

**Comment by RomanBredehoft** [comment]
yes that's the case @bcm-at-zama !

**Comment by bcm-at-zama** [comment]
Isn't it easier and faster to run `make licenses` yourself locally?

**Comment by fd0r** [comment]
If you can do it locally then yes ofc

But for Apple users since we enforce that Linux licenses be up to date it's the best way for you to update the licences.
Also atm it opens a PR to your branch so you can just: 
- push your branch
- launch the action
- merge the PR as soon as available (no need for all checks to pass since you aren't targetting a protected branch)
- pull the update branch to your local
- squash the update licence commit
- push force to origin
- open the PR

**Comment by bcm-at-zama** [comment]
When I was doing PRs, I did it in docker/macintel, and it was ok. If you're happy with this new method, all good too

**Comment by bcm-at-zama** [comment]
sure!

**Pull request merged**